addons/l10n_in/models/account_invoice.py

```diff
@@ -140,6 +165,129 @@ def build_warning(record, action_name, message, views, domain=False):
+
+        def get_deemed_export_section(move):
+            if (
+                is_invoice(move) and
+                move.l10n_in_gst_treatment == 'deemed_export' and
+                has_tags(gst_tags)
+            ):
+                return 'deemed_export'
+
+        def get_nil_rated_section(move):
+            if (
+                move.l10n_in_gst_treatment not in ('overseas', 'special_economic_zone') and
+                has_tags(nil_tags)
+            ):
+                return 'nil_rated'
+
+        def get_cdnr_section(move):
+            if not (is_invoice(move) or move.is_inbound()):
+                if move.l10n_in_gst_treatment in ('regular', 'composition', 'uin_holders'):
+                    if has_tags(gst_rc_tags):
+                        return 'cdnr_rcm'
+                    elif has_tags(gst_tags):
+                        return 'cdnr_regular'
+                elif move.l10n_in_gst_treatment == 'special_economic_zone' and has_tags(export_tags):
+                    return 'cdnr_sez'
+                elif move.l10n_in_gst_treatment == 'deemed_export' and has_tags(gst_tags):
+                    return 'cdnr_deemed_export'
+
+        def get_cdnur_section(move):
+            if not (is_invoice(move) or move.is_inbound()):
+                if (move.l10n_in_gst_treatment in ('unregistered', 'consumer') and
+                    has_tags(gst_tags) and
+                    get_transaction_type(move) == 'inter_state' and
+                    (
+                        (move.debit_origin_id and move.debit_origin_id.amount_total > 250000) or
+                        (move.reversed_entry_id and move.reversed_entry_id.amount_total > 250000) or
+                        (not move.reversed_entry_id and not move.is_inbound())
+                    )):
+                    return 'cdnur_b2cl'
+                elif move.l10n_in_gst_treatment == 'overseas':
+                    if has_tags(export_tags):
+                        return 'cdnur_exp_wp'
+                    elif has_tags(igst_lut_tag):
+                        return 'cdnur_exp_wop'
+
+        indian_moves = self.filtered(lambda m: m.country_code == 'IN' and m.is_sale_document(include_receipts=True) and m.state == 'posted')
+        (self - indian_moves).l10n_in_gstr_mapping = False
+        if not indian_moves:
+            return
+        gst_tags = self.env.ref('l10n_in.tax_tag_base_sgst') + self.env.ref('l10n_in.tax_tag_base_cgst') + self.env.ref('l10n_in.tax_tag_base_igst')
+        gst_rc_tags = self.env.ref('l10n_in.tax_tag_base_sgst_rc') + self.env.ref('l10n_in.tax_tag_base_cgst_rc') + self.env.ref('l10n_in.tax_tag_base_igst_rc')
+        nil_tags = self.env.ref('l10n_in.tax_tag_exempt') + self.env.ref('l10n_in.tax_tag_nil_rated') + self.env.ref('l10n_in.tax_tag_non_gst_supplies')
+        export_tags = self.env.ref('l10n_in.tax_tag_zero_rated') + self.env.ref('l10n_in.tax_tag_base_igst')
+        igst_lut_tag = self.env.ref('l10n_in.tax_tag_igst_lut')
+
```

Comment:
```suggestion
gstr_mapping_fun = {
    "b2b_regular": is_b2b_section
    "b2b_rcm": is_b2b_rcm
    "b2cl" is_b2cl
    "b2cs": is_b2cs
    "exp_wp": is_exp_wp
    "exp_wop": is_exp_wop
    ...
    ...
    ...
}
```
Create map where the key is GSTR-1 sections and the left side is a method